tests/conftest.py

```diff
@@ -49,13 +49,25 @@ def tls_ca_certificate_private_key_encrypted_path(tls_certificate_authority):
 
 
 @pytest.fixture
-def tls_certificate_pem_path(tls_certificate: trustme.LeafCert):
+def tls_certificate_private_key_path(tls_certificate: trustme.CA):
+    with tls_certificate.private_key_pem.tempfile() as private_key:
+        yield private_key
+
+
+@pytest.fixture
+def tls_certificate_key_and_chain_path(tls_certificate: trustme.LeafCert):
     with tls_certificate.private_key_and_cert_chain_pem.tempfile() as cert_pem:
         yield cert_pem
 
 
 @pytest.fixture
-def tls_ca_ssl_context(tls_certificate: trustme.LeafCert) -> ssl.SSLContext:
-    ssl_ctx = ssl.SSLContext()
-    tls_certificate.configure_cert(ssl_ctx)
+def tls_certificate_server_cert_path(tls_certificate: trustme.LeafCert):
+    with tls_certificate.cert_chain_pems[0].tempfile() as cert_pem:
+        yield cert_pem
+
+
+@pytest.fixture
+def tls_ca_ssl_context(tls_certificate_authority: trustme.CA) -> ssl.SSLContext:
+    ssl_ctx = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+    tls_certificate_authority.configure_trust(ssl_ctx)
     return ssl_ctx
```

Comment:
yeah that's way better, I did not know a vanilla context had by default CERT_NONE !